crates/services/consensus_module/poa/src/service.rs

```diff
@@ -387,6 +402,70 @@ where
         Ok(())
     }
 
+    async fn produce_predefined_block(
+        &mut self,
+        predefined_block: &Block,
+        chain_id: &ChainId,
+    ) -> anyhow::Result<()> {
+        tracing::info!("Producing predefined block");
+        let last_block_created = Instant::now();
+        // verify signing key is set
+        if self.signing_key.is_none() {
+            return Err(anyhow!("unable to produce blocks without a consensus key"))
+        }
+
+        // Ask the block producer to create the block
+        let (
+            ExecutionResult {
+                block,
+                skipped_transactions,
+                tx_status,
+                events,
+            },
+            changes,
+        ) = self
+            .block_producer
+            .produce_predefined_block(predefined_block)
+            .await?
+            .into();
+
+        if !skipped_transactions.is_empty() {
+            tracing::error!("During block production got invalid transactions");
+            let txs = predefined_block.transactions();
+
+            for (tx_id, err) in skipped_transactions {
+                let maybe_tx = txs.iter().find(|tx| tx.id(chain_id) == tx_id);
+                if let Some(tx) = maybe_tx {
+                    tracing::error!(
+                "During block production got invalid transaction {:?} with error {:?}",
+                tx,
+                err
+            );
+                }
+            }
+        }
+        // Sign the block and seal it
+        let seal = seal_block(&self.signing_key, &block)?;
+        let sealed_block = SealedBlock {
+            entity: block,
+            consensus: seal,
+        };
+        // Import the sealed block
+        self.block_importer
+            .commit_result(Uncommitted::new(
+                ImportResult::new_from_local(sealed_block.clone(), tx_status, events),
+                changes,
+            ))
+            .await?;
+
+        // Update last block time
+        self.last_height = *sealed_block.entity.header().height();
+        self.last_timestamp = sealed_block.entity.header().time();
+        self.last_block_created = last_block_created;
```

Comment:
Because you don't set the timer here, I think after applying predefined blocks the server will halt in the case of `Interval` trigger more

Reply:
Maybe we could somehow reuse main part of the `produce_block` function

Reply:
It will be removed during https://github.com/FuelLabs/fuel-core/issues/1917